web/src/search_suggestion.ts

```diff
@@ -164,14 +153,10 @@ function get_channel_suggestions(last: NarrowTerm, terms: NarrowTerm[]): Suggest
 
     channels = typeahead_helper.sorter(query, channels, (x) => x);
 
-    const regex = typeahead_helper.build_highlight_regex(query);
-    const highlight_query = typeahead_helper.highlight_with_escaping_and_regex;
-
     return channels.map((channel_name) => {
         const prefix = "channel";
-        const highlighted_channel = highlight_query(regex, channel_name);
         const verb = last.negated ? "exclude " : "";
-        const description_html = verb + prefix + " " + highlighted_channel;
+        const description_html = verb + prefix + " " + channel_name;
```

Comment:
This drops the escaping on `channel_name` and thus introduces an XSS vulnerability. Never put user-generated input into HTML without escaping.

Reply:
Oh shoot sorry I missed this one, I caught it for the user name. I'll add a test too! Thanks